### PR TITLE
Real world test 1

### PR DESCRIPTION
**Pull request by EdNoepel**
Removed hardcoded random seed, which helped find another LUP management bug last week.  Track when user index last triggered an action to throttle interactions at desired intervals.  Improved the book dump facility to show LP balances and more precision on inflator values.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -239,40 +240,93 @@ def _end_profiling(self):
             )
 
     @staticmethod
-    def dump_book(pool, bucket_math, min_bucket_index=-3232, max_bucket_index=6926, with_headers=True, csv=False) -> str:
+    def validate_book(pool, bucket_math, min_bucket_index=-3232, max_bucket_index=6926):
```
Comment (MikeHathaway) [inline tests/conftest.py]: The max_bucket_index is now 4156: https://github.com/ajna-finance/contracts/blob/develop/src/libraries/BucketMath.sol#L25

File: tests/test_stable_volatile.py [hunk 2]
```diff
@@ -90,166 +120,185 @@ def draw_initial_debt(
...
+    end_time = chain.time() + duration
+    # Update the interest rate
+    interest_rate = update_interest_rate(lenders, pool)
+    chain.sleep(14)
+
+    while chain.time() < end_time:
+        if chain.time() - last_triggered[user_index] > get_time_between_interactions(user_index):
+
+            # Draw debt, repay debt, or do nothing depending on interest rate
+            utilization = pool.getPoolActualUtilization() / 10**18
```
Comment (MikeHathaway) [inline tests/test_stable_volatile.py]: getPoolActualUtilization returns a RAY post inflator-precision update

File: tests/test_stable_volatile.py [hunk 3]
```diff
@@ -90,166 +120,185 @@ def draw_initial_debt(
...
+            try:
+                if interest_rate < 0.10 and utilization < 0.80:  # draw more debt if interest is reasonably low
+                    target_collateralization = 1 / pool.getPoolTargetUtilization() * 10**18
```
Comment (MikeHathaway) [inline tests/test_stable_volatile.py]: `getPoolTargetUtilization` returns a RAY

File: tests/test_stable_volatile.py (hunk 2 above)
Comment (EdNoepel) [reply tests/test_stable_volatile.py]: Addressed in e5133b3a96a4ce3d245208882e32874d07f4055b

File: tests/test_stable_volatile.py (hunk 3 above)
Comment (EdNoepel) [reply tests/test_stable_volatile.py]: Addressed in e5133b3a96a4ce3d245208882e32874d07f4055b

File: tests/conftest.py (hunk 1 above)
Comment (EdNoepel) [reply tests/conftest.py]: Addressed in e5133b3a96a4ce3d245208882e32874d07f4055b

**Review by MikeHathaway — APPROVED** [review]
LGTM

**Pull request merged**